src/bundle/settings.rs

```diff
@@ -100,23 +99,90 @@ impl Settings {
-            is_release: is_release,
+            package,
+            package_type,
+            target,
+            is_release,
             project_out_directory: target_dir,
-            binary_path: binary_path,
-            bundle_settings: bundle_settings,
+            binary_path,
+            bundle_settings,
         })
     }
 
+    /*
+        The target_dir where binaries will be compiled to by cargo can vary:
+            - this directory is a member of a workspace project
+            - overridden by CARGO_TARGET_DIR environment variable
+            - specified in build.target-dir configuration key
+            - if the build is a 'release' or 'debug' build
+
+        This function determines where 'target' dir is and suffixes it with 'release' or 'debug'
+        to determine where the compiled binary will be located.
+    */
+    fn get_target_dir(project_root_dir: &PathBuf, target: &Option<(String, TargetInfo)>, is_release: bool)
+                      -> PathBuf {
+        let mut path = project_root_dir.join("target");
+        if let &Some((ref triple, _)) = target {
+            path.push(triple);
+        }
+        path.push(if is_release { "release" } else { "debug" });
+        path
+    }
+
+    /*
+        workspace section spec of Manifest format:
+        https://doc.rust-lang.org/cargo/reference/manifest.html#the-workspace-section
+
+        Determining if part of workspace, and which:
+            "The package.workspace manifest key (described above) is used in member crates to
+             point at a workspace's root crate.
+
+             If this key is omitted then it is inferred to be the first crate whose manifest contains
+             [workspace] upwards in the filesystem."
+
+             The Cargo.toml (found walking up the FS) could now have a workspace.exclude entry
+             for this crate, meaning that this crate does not below to a workspace.
+
+             "Note that members of the workspaces listed explicitly will also have their path
+             dependencies included in the workspace"
+
+             QUESTION: If no workspace root directory is specified by the package.workspace key
+             do we just look in the parent directory for a Cargo.toml defining a workspace we belong
+             to, or do we have to walk up the file system until "/" to see if we can find a
+             Cargo.toml?
+    */
+    fn get_workspace_dir(current_dir: &PathBuf) -> PathBuf {
+        let mut dir = current_dir.clone();
+        while dir.pop() {
+            let set = Self::cargo_settings(&dir);
+            if set.is_ok() {
+                return dir;
+            }
+        }
+
+        // Nothing found walking up the file system, return the starting directory
+        current_dir.clone()
+    }
+
+    /*
+        Try to load a set of CargoSettings from a "Cargo.toml" file in the specified directory
+    */
+    fn cargo_settings(dir: &PathBuf) -> ::Result<CargoSettings> {
```

Comment:
Seems like this would make more sense as a method of `CargoSettings`, rather than `Settings`.  Maybe `CargoSettings::load()`?

Reply:
Yes, off course! Makes sense.